extensions/amp-iframe/1.0/component.js

```diff
@@ -29,6 +35,46 @@ export function Iframe({
   const isIntersectingRef = useRef(null);
   const containerRef = useRef(null);
 
+  const viewabilityCb = (entries) => {
+    const iframe = iframeRef.current;
+    if (!iframe) {
+      return;
+    }
+    postMessage(
+      iframe,
+      MessageType.INTERSECTION,
+      dict({'changes': entries.map(cloneEntryForCrossOrigin)}),
+      '*'
+    );
+  };
+
+  useEffect(() => {
+    const iframe = iframeRef.current;
+    if (!iframe) {
+      return;
+    }
+    const win = toWin(iframe.ownerDocument.defaultView);
+    let observer;
+    win.addEventListener('message', (event) => {
+      // incl base param to URL constructor for relative URLs used in storybooks
+      const {origin} = new URL(src, win.location.origin);
```

Comment:
This is the wrong `win`. The iframe window's origin cannot be inspected because it's cross-origin. You're attempting to the the parent window's origin here, which means it would only work for same-origin iframes.

Reply:
To clarify, I was trying make sure we can handle relative URLs. Per MDN docs:
```
// Base urls
let m = 'https://example.com';
let b = new URL(m); 
new URL('http://www.example.com', b);  // => 'http://www.example.com/'
new URL('/en-US/docs', b);   // => 'https://example.com/en-US/docs'
```
Glad it's not required anymore.